src/main/java/com/sarality/sync/APISyncExecutor.java

```diff
@@ -26,6 +30,9 @@
   private final APISyncRequestGenerator<S, R> requestGenerator;
   private final APICallExecutor<S, R> apiExecutor;
   private TaskProgressListener<SyncProgressCount> progressListener;
+  private List<SyncErrorData> syncErrorList = new ArrayList<>();
```

Comment:
I see your point about SyncError vs SyncErrorData. Makes sense to use SyncError here when you do the refactoring in the future.